### PR TITLE
update docs for sparse tensor

### PR DESCRIPTION
**Pull request by weiyangfb**
- update docs examples at sparse tensor after print format changed
- update example to create empty sparse tensor:
```
>>> torch.sparse_coo_tensor(torch.LongTensor(size=[1,0]), [], torch.Size([1]))
tensor(indices=tensor([], size=(1, 0)),
       values=tensor([], size=(0,)),
       size=(1,), nnz=0, layout=torch.sparse_coo)
```

@zou3519 @SsnL @yf225

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -4215,32 +4215,31 @@ def parse_kwargs(desc):
...
-    tensor([])
+    tensor(indices=tensor([[0, 1, 1],
+                           [2, 0, 2]]),
+           values=tensor([3., 4., 5.]),
+           device='cuda:0', size=(2, 4), nnz=3, dtype=torch.float64,
+           layout=torch.sparse_coo)
+
+    # Create an empty tensor (of size (1,))
+    # an empty sparse tensor has sparseDims = 1 and denseDims = 0, and indices size = (1, 0)
+    >>> S = torch.sparse_coo_tensor(torch.LongTensor(size=[1,0]), [], torch.Size([1]))
```
Comment (SsnL) [inline torch/_torch_docs.py]: nit: space in `1,0`

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -4215,32 +4215,31 @@ def parse_kwargs(desc):
...
-    and values:
-    tensor([])
+    tensor(indices=tensor([[0, 1, 1],
+                           [2, 0, 2]]),
+           values=tensor([3., 4., 5.]),
+           device='cuda:0', size=(2, 4), nnz=3, dtype=torch.float64,
+           layout=torch.sparse_coo)
+
+    # Create an empty tensor (of size (1,))
+    # an empty sparse tensor has sparseDims = 1 and denseDims = 0, and indices size = (1, 0)
```
Comment (SsnL) [inline torch/_torch_docs.py]: nit: use `sparse_dim` and `dense_dim`. their names are changing in #11253

**Review by ssnl — APPROVED** [review]

File: torch/_torch_docs.py [hunk 3]
```diff
@@ -4215,32 +4215,31 @@ def parse_kwargs(desc):
...
-    tensor([], dtype=torch.int64)
-    and values:
-    tensor([])
+    tensor(indices=tensor([[0, 1, 1],
+                           [2, 0, 2]]),
+           values=tensor([3., 4., 5.]),
+           device='cuda:0', size=(2, 4), nnz=3, dtype=torch.float64,
+           layout=torch.sparse_coo)
+
+    # Create an empty tensor (of size (1,))
```
Comment (SsnL) [inline torch/_torch_docs.py]: `(1,)` is not empty

File: torch/_torch_docs.py (hunk 2 above)
Comment (SsnL) [reply torch/_torch_docs.py]: You talked nothing about number of nonzero elements. Can you expand the sentence a bit? It is not very obvious that indices is still 2d.

Also, other examples on empty tensors with different densities (i.e., `dense_dim`) would be helpful in explaining.

File: torch/_torch_docs.py (hunk 1 above)
Comment (SsnL) [reply torch/_torch_docs.py]: nit: use `torch.tensor` ctor, or others like `empty` and `randn`. We shouldn't advocate `LongTensor` ctors anymore.

**Review by ssnl — CHANGES_REQUESTED** [review]

File: torch/_torch_docs.py (hunk 2 above)
Comment (weiyangfb) [reply torch/_torch_docs.py]: I see, I am adding to the docs with invariants when creating an empty sparse tensor, and also an example of empty sparse tensor with `dense_dim = 1`

**Comment by weiyangfb** [comment]
@SsnL is this ready to go?

File: torch/_torch_docs.py [hunk 4]
```diff
@@ -4211,36 +4211,49 @@ def parse_kwargs(desc):
...
-    tensor([])
+    tensor(indices=tensor([[0, 1, 1],
+                           [2, 0, 2]]),
+           values=tensor([3., 4., 5.]),
+           device='cuda:0', size=(2, 4), nnz=3, dtype=torch.float64,
+           layout=torch.sparse_coo)
+
+    # Create an empty sparse tensor with the following invariants:
+    #   1. sparse_dim + dense_dim = len(SparseTensor.shape)
+    #   2. SparseTensor._indices().shape = (spares_dim, nnz)
```
Comment (SsnL) [inline torch/_torch_docs.py]: nit: sparse not spares

File: torch/_torch_docs.py [hunk 5]
```diff
@@ -4211,36 +4211,49 @@ def parse_kwargs(desc):
...
+    #   3. SparseTensor._values().shape = (nnz, SparseTensor.shape[sparse_dim:])
+    #
+    # For instance, to create an empty sparse tensor with nnz = 0, dense_dim = 0 and
+    # sparse_dim = 1 (hence indices is a 2D tensor of shape = (1‚ 0))
+    >>> S = torch.sparse_coo_tensor(torch.tensor([]).reshape(1, 0), [], [1])
```
Comment (SsnL) [inline torch/_torch_docs.py]: Why is there a reshape? Can you just use one of the ctors that take in shape?

File: torch/_torch_docs.py [hunk 6]
```diff
@@ -4211,36 +4211,49 @@ def parse_kwargs(desc):
...
+    tensor(indices=tensor([], size=(1, 0)),
+           values=tensor([], size=(0,)),
+           size=(1,), nnz=0, layout=torch.sparse_coo)
+
+    # and to create an empty sparse tensor with nnz = 0, dense_dim = 1 and
+    # sparse_dim = 1
+    >>> S = torch.sparse_coo_tensor(torch.tensor([]).reshape(1, 0),
```
Comment (SsnL) [inline torch/_torch_docs.py]: Same here

**Review by ssnl — APPROVED** [review]

**Pull request closed without merging**